src/sonic-py-common/sonic_py_common/multi_asic.py

```diff
@@ -321,6 +321,30 @@ def is_port_channel_internal(port_channel, namespace=None):
 
     return False
 
+# Allow user to get a set() of back-end interface and back-end LAG per namespace
+# default is getting it for all name spaces if no namespace is specified
+def get_back_end_interface_set(namespace=None):
+    bk_end_intf_list =[]
+    if not is_multi_asic():
+        return None
+
+    port_table = get_port_table(namespace)
+    for port, info in port_table.items():
+        if PORT_ROLE in info and info[PORT_ROLE] == INTERNAL_PORT:
+            bk_end_intf_list.append(port)
+
+        ns_list = get_namespace_list(namespace)
+        for ns in ns_list:
+            config_db = connect_config_db_for_ns(ns)
+            port_channels = config_db.get_table(PORT_CHANNEL_CFG_DB_TABLE)
+            for port_channel, lag_info in port_channels.items():
+                if 'members' in lag_info:
+                    members = lag_info['members']
+                    if members[0] in bk_end_intf_list:
+                        bk_end_intf_list.append(port_channel)
```

Comment:
is this intended to be inside for loop. This will get called for every port in the port_table.

Reply:
@smaheshm Yes this is intended. for each namespace this needs to be performed if user use default option which is for all name spaces.  each namespace has its own LAG DB.

Reply:
I meant this for loop is inside the port for loop.

Reply:
@smaheshm Thanks for spotting it.  When I copied over it added extra spacing.  Let me fix that.

Reply:
@gechiang we can use any function. I got that as one of review comment .
Refer: https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-config-engine/minigraph.py#L756

Also we can consolidate both going forward however not in this PR.

Reply:
@abdosi I took a look at the link you provided.  I see the logic is to check if there are any LAG members being a back-end interface then the LAG is a back-end port channel.  I have a question.  Can we have a case where the port channel is formed with front-end ports and back-end ports mixed as LAG member?  I do not think this can ever be a valid usage of such port channel right?  If that is the case, then my change which is also same as what is already done in the method " is_port_channel_internal()" should be faster right?  It just have to check the very first member and result can already be determined. Think about the case where it is a true front-end port channel with many lag members.  The method you pointed out will require compare ALL of the lag members in order to determine that it is NOT a back-end port channel. performance wise it is not as efficient as what my check can achieve...  Also, that method assumed there are alwasy some members for a port channel.  that may not be always true as some one may have configured an empty port channel which in that case that code may crash as it assumed "member" is present...

Reply:
@gechiang any function should return immediately like first member check. Basically it is not from optimized perspective but better python readability.

Reply:
@abdosi I can replace that block of code: 

            for port_channel, lag_info in port_channels.items():
                if 'members' in lag_info:
                    members = lag_info['members']
                    if members[0] in bk_end_intf_list:
                        bk_end_intf_list.append(port_channel)


with the following one line code:
```
bk_end_intf_list.extend([port_channel for port_channel, lag_info in port_channels.items()\
                                if 'members' in lag_info and lag_info['members'][0] in bk_end_intf_list])
```
If you feel this is more readable...  I actually prefer the original "c" style for clarity although few more lines that does the same thing...  

As for the any() function it is not efficient for the case when LAG contains all front-end ports where it will have to evaluate all the lag members in order to determine that LAG should not be included so I will not use that.

Reply:
Code should be idiomatic. And in general readability is preferred over efficiency especially when the efficiency doesn't make a big difference. Easier to maintain code.

Reply:
I have updated the code to use this one line list expansion and added comments so it should allow reader to understand what are the criteria that determine a LAG is a back-end LAG...